data/pagination.go

```diff
@@ -0,0 +1,171 @@
+package data
+
+import (
+	"context"
+	"net/url"
+	"strconv"
+
+	"github.com/ONSdigital/dp-frontend-models/model"
+	errs "github.com/ONSdigital/dp-frontend-search-controller/apperrors"
+	"github.com/ONSdigital/dp-frontend-search-controller/config"
+	"github.com/ONSdigital/log.go/log"
+)
+
+// LimitOptions contains all available limit parameter values
+var LimitOptions = []int{
+	10,
+	25,
+	50,
+}
+
+// reviewPagination reviews page and limit values and sets limit and page values to queryParams
+func reviewPagination(ctx context.Context, cfg *config.Config, urlQuery url.Values, validatedQueryParams *SearchURLParams) error {
+	limit := getLimitFromURLQuery(ctx, cfg, urlQuery)
+	validatedQueryParams.Limit = limit
+
+	page := getPageFromURLQuery(ctx, cfg, urlQuery)
+	validatedQueryParams.CurrentPage = page
+
+	offset, err := getOffset(ctx, cfg, page, limit)
+	if err != nil {
+		log.Event(ctx, "unable to get offset", log.Error(err), log.ERROR)
+		return err
+	}
+	validatedQueryParams.Offset = offset
+
+	return nil
+}
+
+func getLimitFromURLQuery(ctx context.Context, cfg *config.Config, query url.Values) int {
+	defaultLimitStr := strconv.Itoa(cfg.DefaultLimit)
+
+	limit, err := strconv.Atoi(query.Get("limit"))
+	if err != nil {
+		log.Event(ctx, "unable to convert search limit to int - set to default "+defaultLimitStr, log.INFO)
```

Comment:
Concatenating strings can get ugly and requires casting of types to string. This gets especially horrific when you are concatenating multiple values. I think a more Go-centric way to achieve the same would be to use `fmt` with placeholders:
```
log.Event(ctx, fmt.Sprintf("unable to convert search limit to int - set to default %d", defaultLimit), log.INFO)
```
If you do this you can remove the string var on line 40.